### PR TITLE
Remove stuck weapon penalty 

### PR DESCRIPTION
**Pull request by mugling**
Four flags and 200 lines of code for the badly designed stuck weapon penalty isn't sensible and is an obstacle to a better melee system. In particular it prevents weapons having both `cutting` and `piercing` stats. This makes it hard to design items such as the `halberd`. This PR drops support for the penalty.

The next PR will update the JSON data to specify explicit values for `piercing` instead of relying on the `STAB` and `SPEAR` flags. Longer term the different damage types will start to diverge and have varying effectiveness against different monsters with the game automatically selecting the best available option.

### CODE REVIEW

**Comment by mugling** [comment]
Added bonus: fixes #14882 which is currently blocking `0.D`.

Ready

File: doc/JSON_FLAGS.md [hunk 1]
```diff
@@ -780,19 +780,15 @@ Some armor flags, such as `WATCH` and `ALARMCLOCK` are compatible with other ite
 ### Flags
 
 - ```ALWAYS_TWOHAND``` Item is always wielded with two hands. Without this, the items volume and weight are used to calculate this.
-- ```CHOP``` Does cutting damage, with a high chance of getting stuck.
 - ```DIAMOND``` Diamond coating adds 30% bonus to cutting damage
 - ```MESSY``` Resistant to getting stuck in a monster. Potentially cause more gore in the future?
```
Comment (Coolthulhu) [inline doc/JSON_FLAGS.md]: Needs description change. It does create more gore (especially fun when smashing corpses), but no longer affects sticking.

File: src/melee.cpp [hunk 2]
```diff
@@ -1624,69 +1555,8 @@ std::string player::melee_special_effects(Creature &t, damage_instance &d, const
...
-        if( cutting_penalty >= 50 && !t.is_hallucination() ) {
-            dump << string_format(_("Your %1$s gets stuck in %2$s but you yank it free!"), weapon.tname().c_str(),
-                                  target.c_str());
-        }
-
-        if( weapon.has_flag("SPEAR") ) {
-            ///\EFFECT_STABBING increases damage when pulling out a stuck spear
-            const int stabbing_skill = get_skill_level( skill_stabbing );
-            d.add_damage( DT_CUT, rng( 1, stabbing_skill ) ); //add some extra damage for pulling out a spear
-            t.mod_moves(-30);
```
Comment (Coolthulhu) [inline src/melee.cpp]: Move draining part of spears is one of the features I really liked.
Still, it wasn't really balanced, especially after spears got their reach attacks.

Might be worth resurrecting in some other form.

**Comment by DangerNoodle** [comment]
I can see this being a good thing, yes. I forget whether or not the Deft trait affects getting weapons stuck.

**Comment by mugling** [comment]
> I can see this being a good thing, yes

Almost all of the effects never trigger

**Comment by DangerNoodle** [comment]
Regarding the stuckin code, or deftness? Because I recall getting a weapon stuck tends to be somewhat common occurrence with 0 skills.

**Comment by mugling** [comment]
Deftness has no effect

File: doc/JSON_FLAGS.md (hunk 1 above)
Comment (mugling) [inline doc/JSON_FLAGS.md]: Fixed

**Comment by mugling** [comment]
> Unrelated, but comical:
> 
>     mugling added some commits 22 minutes from now

Were you not aware that I can predict the future?

**Comment by DangerNoodle** [comment]
> Deftness has no effect

I might look into seeing how to mess with recovery time of missed strikes in a later pull request then. Unless you meant that it had no effects that involved weapons sticking in a foe, which was the original intent of my question.

**Comment by illi-kun** [comment]
@mugling It seems that you support obsolete version of ```PK rebalance``` mod (see https://github.com/pisskop/PKs_Rebalancing for actual version).

Do you have any idea how you (and other active devs) can announce significant changes in code to inform mod maintainers about required changes?

**Comment by mugling** [comment]
To be honest we probably should drop that mod from `core` as it's only going to languish there much like legacy blazemod did.

We could do with a developer announcement thread. Something locked to developers that people could subscribe to notifications on?

**Comment by illi-kun** [comment]
I think it will be ideal solution.

**Comment by mugling** [comment]
The question is do we publish changes ad-hoc or on an episodic (weekly) basis?

**Comment by illi-kun** [comment]
I dunno. ad-hoc is better, I guess (since it's easier to miss something in weekly announces).

**Comment by illi-kun** [comment]
Re: drop the mod from core content.
For doing so, we need to provide a clear link to updated version of mod, so players will able to download & setup it easily (we don't need to enforce players to stop playing with this mod).

@remyroy I heard that a lot of people enjoy using your CDDA launcher. Is it possible to request you about a new feature - downloading fresh versions of mods from dedicated repos? I guess I need to write a feature request to your repo about that (in case you're interested in implementation of this feature)?

**Comment by remyroy** [comment]
@illi-kun Yes, please create an issue about it on https://github.com/remyroy/CDDA-Game-Launcher/issues/new

**Comment by pisskop** [comment]
Ill take the time to update it (pks_rebalancing) and reduce it to just what it promises, if you all dont mind.

**Comment by DangerNoodle** [comment]
I was going to say, Pisskop's mod is more actively maintained that Blazemod is.

**Comment by pisskop** [comment]
Does this allow for the same kind of abstraction as ammo?

Can I for instance create a template for a horse.  Then add 1.2 attack to it for a zombied version, and then make a 0.8 version for fungus-infected version?

**Comment by mugling** [comment]
> I was going to say, Pisskop's mod is more actively maintained that Blazemod is.

Yes I would agree although I suspect your confounding maintenance with expansion. Requiring significant maintenance is a burden on the developers. That the mod author continues to expand the mod is immaterial to this.

> Ill take the time to update it (pks_rebalancing) and reduce it to just what it promises, if you all dont mind.

I set out in #19037 some possible criteria for mod inclusion:

> - reasonably broad appeal
> - an active maintainer
> - acceptable code standards

The third was a problem that I hoped to fix via linting. In particular when I updated `edit-mode` to `copy-from` I could use `jq` to make those changes automatically as that part of your mod had been linted. The ancillary problem was dispute over content - discussion is fine but past a certain point if there is no agreement the options are continue in your own branch or accept the developers decision.

> Does this allow for the same kind of abstraction as ammo?

Yes it will although `relative` and `proportional` are not yet supported (but will be soon).

There are 47 mods in `core` which is an absolutely silly number to try and support with so few developers. No amount of opining or lengthy forum posts are going to change that fact.

The question with PK mod is what does the project as a whole gain by supporting the mod given it's relatively expensive to do so (genuine question, no snark intended)?

**Comment by DangerNoodle** [comment]
> Yes I would agree although I suspect your confounding maintenance with expansion. Requiring significant maintenance is a burden on the developers. That the mod author continues to expand the mod is immaterial to this.

True. Having the mod author around makes it more likely that they will be able to respond to any issues that arise, but does not guarantee it.

**Comment by mugling** [comment]
@remyroy could the auto-updater perhaps support a semi-stable weekly build instead? Do you have any statistics on how many players we have, their geographic distribution and update frequency?

@DangerNoodle the mod author doesn't do any maintenance on the mod - that's all done by me and sometimes @Coolthulhu. We had the same problem with blazemod until it will did that exorcism a while back via the linter.

**Comment by DangerNoodle** [comment]
> @DangerNoodle the mod author doesn't do any maintenance on the mod - that's all done by me and sometimes @Coolthulhu. We had the same problem with blazemod until it will did that exorcism a while back via the linter.

I was unaware of that. My apologies for that.

**Comment by mugling** [comment]
It's a sincere question though what does the mod have to offer. On the one hand mods attract new developers but in our current situation we spend more time resolving issues with mods than actually working on the game.

**Comment by remyroy** [comment]
@mugling Yes. As long as those builds are somewhere I can list them, sure.

**Comment by pisskop** [comment]
If were going to make it possible for 3rd party mods to be used and updated in conjunction with the game updates than there is no reason to keep it there.  Or any large content mod with an owner, really.

**Comment by remyroy** [comment]
@mugling I do not have any statistics on how many players we have, their geographic distribution or their update frequency. Maybe @narc0tiq has some logs on his server since he is the one hosting the actual binary files.

I might be interesting to do another player survey like I did last time (#14273) and include those questions.

**Comment by narc0tiq** [comment]
The live-ish stats at http://dev.narc.ro/awstats/awstats.pl might help, as
far as who downloads what and from where.

I'm not anywhere near a computer, so this is all I have.

**Comment by remyroy** [comment]
The latest version of the CDDA Game Launcher (1.3.5) uses a custom User-Agent header *CDDA-Game-Launcher/[version]* when making requests on *.narc.ro. It should be easy to identify if someone wants to create detailed statistics about it.

**Pull request merged**